Explain the npm verbosity in the node image

Starefossen opened the pull request:
Attempts to explain the rationale behind the `NPM_CONFIG_LOGLEVEL`.

Close: nodejs/docker-node#57
Related: nodejs/docker-iojs#36

Signed-off-by: Hans Kristian Flaatten <hans.kristian.flaatten@dnt.no>

pesho commented:
LGTM

chorrell commented:
LGTM

chorrell commented on README.md:
```diff
@@ -53,6 +53,45 @@ $ docker run -it --rm --name my-running-script -v "$PWD":/usr/src/app -w
 
+## Verbosity
+
+By default Node.js Docker Image has NPM log verbosity set to `verbose` instead
```
Should this be changed to `info` instead of `verbose`?

retrohacker replied:
"By default Node.js" -> "By default the Node.js"

`has NPM log verbosity` -> `has the npm log verbosity`

chorrell commented:
One small nit, but other-wise looks good

retrohacker commented on README.md:
```diff
@@ -53,6 +53,45 @@ $ docker run -it --rm --name my-running-script -v "$PWD":/usr/src/app -w
+By default Node.js Docker Image has NPM log verbosity set to `verbose` instead
+of the default `warn`. This is because of the way Docker is isolated from the
+host operating system and you are not guaranteed to be able to retrieve the
+`npm-debug.log` file when something breaks.
```
Should we be a bit more explicit? Something like: "When `npm install` fails, it writes it's verbose log file to a log file on the filesystem. If npm fails during an install when building an image with `docker build`, this log file will be lost when the Docker image fails to build and the layer is thrown away. The Docker Working Group choose to be overly verbose during a build to provide an audit trail when an install fails."

(this is me thinking out loud :smile: )

yosifkit replied:
The layer won't be thrown away on a failed docker build; the container is still there in `docker ps -a`. It is just not easy to jump inside the stopped container. `docker cp`, `docker inspect`, and `docker logs` are all we can really do to debug it.

Maybe we'll get the ability to jump into a stopped container: https://github.com/docker/docker/pull/19746.

retrohacker commented:
I think the technical sections are :fire:

I think we need to explain the rationale behind the decision in terms newcomers to both Docker and Node will understand while striking a balance with being technically correct.

Starefossen commented:
I have incorporated the feedback and added a section to explain the rationale behind the decision of changing the verbosity level of npm. Feel free to nitpick 😄

chorrell commented:
LGTM!

retrohacker commented:
Merging this as it's gone 5 months without objection
